queries.sql

```diff
@@ -127,4 +127,60 @@ SELECT
 FROM
     animals
 GROUP BY species, date_of_birth
-HAVING date_of_birth BETWEEN '1990-01-01' AND '2000-01-01'
\ No newline at end of file
+HAVING date_of_birth BETWEEN '1990-01-01' AND '2000-01-01'
+
+
+-- Day 3
+
+-- Write queries (using JOIN) to answer the following questions:
+-- What animals belong to Melody Pond?
+
+SELECT name
+FROM animals
+JOIN owners
+ON animals.owner_id = 4;
+
+-- List of all animals that are pokemon (their type is Pokemon).
+
+SELECT animals.name
+FROM animals
+JOIN species
+ON animals.species_id = 1;
+
+-- List all owners and their animals, remember to include those that don't own any animal.
+
+SELECT animals.name,owners.full_name
+FROM owners
+LEFT JOIN animals ON owners.id = animals.owner_id;
+
+-- How many animals are there per species?
+
+SELECT species.name, COUNT(animals.id) AS num_animals
+FROM species
+LEFT JOIN animals ON species.id = animals.species_id
+GROUP BY species.name;
```

Comment:
- [ ] The result for this query state that `5 animals have Pokemon as their species`, but from the data we only have 4 animal that belongs to the Pokemon species, and the animals include;
`Pikachu`, `Blossom`, `Charmander`, and `Squirtle`

Reply:
This is becouse the animal "Ditto" also have a species_id of 1. because we have set the species id with the following code.
UPDATE animals
SET species_id = 
    CASE 
        WHEN name LIKE '%mon' THEN 2
        ELSE 1
    END;